src/deploy/index.ts

```diff
@@ -56,9 +56,10 @@ export const deploy = async function (
   const postdeploys: Chain = [];
   const startTime = Date.now();
 
-  if (previews.frameworkawareness && targetNames.includes("hosting")) {
+  if (targetNames.includes("hosting")) {
```

Comment:
That sounds right to me. None of the framework awareness stuff should work if we don't have a `source` path. 

What I'm a tiny bit afraid of is that this _will_ break someone who _may_ have set `frameworkawareness` but never set `source`. I'm not sure if I want to consider it a breaking change because it _is_ something someone opted into, but that's the consideration in my mind.